azurelinuxagent/common/future.py

```diff
@@ -24,3 +32,43 @@
 
 else:
     raise ImportError("Unknown python version: {0}".format(sys.version_info))
+
+
+def get_linux_distribution(get_full_name, supported_dists):
+    """Abstract platform.linux_distribution() call which is deprecated as of
+       Python 3.5 and removed in Python 3.7"""
+    try:
+        supported = platform._supported_dists + (supported_dists,)
+        osinfo = list(
+            platform.linux_distribution(
+                full_distribution_name=get_full_name,
+                supported_dists=supported
+            )
+        )
+        if not osinfo or osinfo == ['','','']:
+            return get_linux_ditribution_from_distro(get_full_name)
+        full_name = platform_module.linux_distribution()[0].strip()
+        osinfo.append(full_name)
+    except Exception:
+        return get_linux_ditribution_from_distro(get_full_name)
```

Comment:
Might want to check the exception details before forwarding the call, or something like this instead:
```
if 'linux_distribution' in dir(platform)
```